distributed/tests/test_scheduler.py

```diff
@@ -422,6 +423,10 @@ async def test_delete(c, s, a):
         await asyncio.sleep(0.01)
         assert time() < start + 5
 
+    assert x.key not in s.tasks
+
+    s.report_on_key(key=x.key)
```

Comment:
Note that we have to confirm that `x.key` is no longer in `s.tasks`. Otherwise it would just grab the `TaskState` object from there even with the `key` specified. IOW now we are sure this will test the case where no `TaskState` can be found and the `key` must be used